src/configparser.cpp

```diff
@@ -388,19 +388,20 @@ bool ParseStack::ParseExec(const std::string& name, int flags, const std::string
 	return ok;
 }
 
-bool ConfigTag::readString(const std::string& key, std::string& value, bool allow_lf)
-{
 #ifdef __clang__
```

Comment:
I had to move the Clang checks here too because:

1. Clang also defines __GNUC__
2. Older GCCs don't allow diagnostics within functions (error: #pragma GCC diagnostic not allowed inside functions).